cdc/sorter/leveldb/cleaner.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package leveldb
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/sorter/encoding"
+	"github.com/pingcap/ticdc/cdc/sorter/leveldb/message"
+	"github.com/pingcap/ticdc/pkg/actor"
+	actormsg "github.com/pingcap/ticdc/pkg/actor/message"
+	"github.com/pingcap/ticdc/pkg/config"
+	"github.com/syndtr/goleveldb/leveldb"
+	lutil "github.com/syndtr/goleveldb/leveldb/util"
+	"go.uber.org/zap"
+	"golang.org/x/time/rate"
+)
+
+// CleanerActor is an actor that can clean up table data asynchronously.
+type CleanerActor struct {
+	id       actor.ID
+	db       *leveldb.DB
+	wbSize   int
+	closedWg *sync.WaitGroup
+
+	limiter *rate.Limiter
+	router  *actor.Router
+}
+
+var _ actor.Actor = (*CleanerActor)(nil)
+
+// NewCleanerActor returns a cleaner actor.
+func NewCleanerActor(
+	ctx context.Context, id int, db *leveldb.DB, router *actor.Router,
```

Comment:
It looks like we have some parameters that are not used. Are they reserved parameters?

Reply:
No, nice catch!

Reply:
Do we need to keep the ctx?

Reply:
Addressed, thanks!